source/fuzz/fuzzer_pass_swap_functions.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 Shiyu Liu
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_pass_swap_functions.h"
+
+#include "source/fuzz/fuzzer_context.h"
+#include "source/fuzz/transformation_swap_two_functions.h"
+
+namespace spvtools {
+namespace fuzz {
+
+FuzzerPassSwapFunctions::FuzzerPassSwapFunctions(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, transformation_context, fuzzer_context,
+                 transformations) {}
+
+void FuzzerPassSwapFunctions::Apply() {
+  // Collect all function ids in a module.
+  std::vector<uint32_t> function_ids;
+  for (auto& function : *GetIRContext()->module()) {
+    function_ids.emplace_back(function.result_id());
+  }
+
+  size_t id_size = function_ids.size();
+  // Iterate through every combination of id i & j where i!=j.
+  for (size_t i = 0; i < id_size - 1; ++i) {
```

Comment:
If SPIR-V module has no functions, the expression `id_size - 1` will overflow.

Reply:
Although, I admit this is rather unlikely situation :)

Reply:
@sliu-UIUC It would indeed be good to rectify this. Can you do it in a follow-up PR, @sliu-UIUC ?

Reply:
> If SPIR-V module has no functions, the expression `id_size - 1` will underflow.

That's a great catch. I wonder if we use something like this: 
`
for (size_t i = 1; i < id_size ; ++i) {
    size_t tmp = i-1; 
    ...
}
`
would resolve this issue for good.

Reply:
You can just do `for (size_t i = 0; i < id_size; ++i)` in the outer loop. Also, it's ok to replace `id_size` with `function_ids.size()` (i.e. if you want to).

Reply:
> @sliu-UIUC It would indeed be good to rectify this. Can you do it in a follow-up PR, @sliu-UIUC ?

Hi sir this question might sound trivial, but is there a specific format (i.e., title, content) that I need to have for a follow up? 
I assume that the content should quote this issue and explain the problems we are addressing, and the title will be meantion its a follow-up?

Reply:
Good question! Just describing the issue in the title (e.g. `spirv-fuzz: Fix underflow in <transformation name>`) and providing a link to the comment in the description (https://github.com/KhronosGroup/SPIRV-Tools/pull/4236#discussion_r621114293) is enough for me :)

Reply:
> Good question! Just describing the issue in the title (e.g. `spirv-fuzz: Fix underflow in <transformation name>`) and providing a link to the comment in the description ([#4236 (comment)](https://github.com/KhronosGroup/SPIRV-Tools/pull/4236#discussion_r621114293)) is enough for me :)

Hi sir thanks for the feedback. I have addressed the issues you mentioned here: #4253.